libcst/metadata/scope_provider.py

```diff
@@ -536,27 +535,49 @@ def accesses(self) -> Accesses:
         return Accesses(self._accesses)
 
 
+class BuiltinScope(Scope):
+    def __init__(self, globals: Scope) -> None:
+        self.globals: Scope = globals
+        super().__init__(parent=self)
+        for name in dir(builtins):
+            self._assignments[name].add(BuiltinAssignment(name, self))
+
+    def __contains__(self, name: str) -> bool:
+        return name in self._assignments and len(self._assignments[name]) > 0
+
+    def __getitem__(self, name: str) -> Set[BaseAssignment]:
+        if name in self._assignments:
+            return self._assignments[name]
+        return set()
+
+    def record_assignment(self, name: str, node: cst.CSTNode) -> None:
+        pass
```

Comment:
IMO this should raise. We should never add more assignments to the builtin scope.

Reply:
SGTM